File: eau05.py

"""
Créez un programme qui affiche le premier nombre premier supérieur au nombre donné en argument.

Exemples d’utilisation :
$> python exo.py 14
17
$>

Afficher -1 si le paramètre est négatif ou mauvais.

"""
#n=int(input("entré un argument en nombre avoir le nombre premier "))

def is_prime(n):    
  n=14
  if not isinstance(n, int):
    return False
  if n % 2 == 0:
    return False
  for i in range(3, int(math.sqrt(n)) + 1):
    if n % i == 0:
      return False
  return True






def get_next_prime(n):
  i = n + 1
  while True:
    if is_prime(i) == False:
      return i
    else:
      i+= 1
  else:
    return None

      

def main():
  n=int(input ("entrez un nombre:  "))
  next_prime = get_next_prime(n)
  print("Le nombre supérieur à ", n, "est", next_prime)


print(main(14)) 

File: eau05.js
/* Créez un programme qui affiche le premier nombre premier supérieur au nombre donné en argument.
Exemples d’utilisation :
$> python exo.py 14
17
$>
Afficher -1 si le paramètre est négatif ou mauvais.
 */

//Parsing 
function getArguments() {
    const arguments = process.argv.slice(2);
    return arguments;
}

//gestiond'erreur
function isValidArguments(arguments) { 
    if (arguments.length <= 0 && arguments.length  > 1) {
        console.error("ce script nécéssite un argument")
    return false;  
    }
    return true;
}

function areAllNumbers (arguments){
    for (let i = 0; i < arguments.length; i++){
        if (isNaN(arguments[i])){
            console.error("ce script nécessite un  nombre en arguments")
            return false;
        }
    }
    return true;
}

// resolution 
function isPrime(number) {
    if (number <= 1) return false;
    for (let i = 2; i <= Math.sqrt(number); i++){
        if  ( number % i === 0 ) return false;
            return false; // n'est divisible pas i donc pas premier
        
    }
    return true; //n ets premier
}

//trouver le ppremier nombre sup à n 
function firstNumberSuperior(number){
    let nombre = number + 1;
    while (true){
        if (nombrePremier(nombre)){
            return nombre; // retourne le premier nombre premier toruvé
        }
        nombre++; //pass eau nombre suivant
    }
}

// affichage
function displayNexfirstNumber (number){
    const arguments = getArguments();
    if (!isValidArguments(arguments) || (!areAllNumbers(arguments)))
        return;
    const n = Number(arguments[0]); //convertir l'argument en nombre 
    const resultat = displayNexfirstNumber(number);
    console.log(`le premier nombre premier supérieur à ${number} est : ${resultat}`);
}

displayNexfirstNumber();

// Remarque romain 
// renomer les 
